### PR TITLE
[Process] Fixes & testNonBlockingNorClearingIteratorOutput

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #17427 |
| License | MIT |
| Doc PR | - |

hasCallback (added on master in #17427) always returns true currently.

### CODE REVIEW

File: src/Symfony/Component/Process/Pipes/WindowsPipes.php [hunk 1]
```diff
@@ -163,14 +163,11 @@ public function close()
     /**
      * Creates a new WindowsPipes instance.
      *
-     * @param Process $process The process
-     * @param $input
-     *
      * @return WindowsPipes
      */
-    public static function create(Process $process, $input)
+    public static function create(Process $process, $input, $haveReadSupport)
```
Comment (javiereguiluz) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: Should `$haveReadSupport` be renamed to `$hasReadSupport`?

Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: that's the name if the var on the constructor and I don't want to rename it there, thus same name here...

Comment (xabbuh) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: and "pipes" is plural, so that looks right here

Comment (xabbuh) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: But we should now remove the `$process` parameter as it is not needed anymore, right?

Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: right, fixed by removing the create method, it's not needed at all

**Pull request merged**